### DISCUSSION TITLE
Why pinta using .dll files instead of .so files?

### DISCUSSION THREAD

**Discussion by visious25** [PintaProject/Pinta#826] [General]
Hi, why pinta using .dll files instead of .so files? The .dll files are vulnerable for viruses.

**Comment by cameronwhite** [comment]
These have a .dll extension because they are C# / dotnet assemblies (e.g. https://learn.microsoft.com/en-us/dotnet/core/deploying/#produce-a-cross-platform-binary)

These are not the same as native shared libraries which typically would have a .so extension on Linux

**Comment by visious25** [comment]
Is there something to change this it in the future?

**Comment by cameronwhite** [comment]
I don't have any plans to change this, because there is no inherent security issue with the choice of file extension, and it's the default behaviour of `dotnet` so doing unusual custom behaviour like this is just a maintenance burden
